### PR TITLE
Basic confirmation modal for sending a letter

### PR DESCRIPTION
**Pull request by shakao**
Adds routes and modal component with text for the four cases (LOB sends/user sends, email to landlord/don't email). 
Some of this will also have to be adjusted when we add handling for preexisting landlord email, but wanted to get the base modal up!

TODO:
- CSS cleanup to match the mocks
- Add the "I don't have this information" checkbox

[sc-8689]
[sc-9374]

### CODE REVIEW

File: frontend/lib/laletterbuilder/letter-builder/send-options.tsx [hunk 1]
```diff
@@ -44,39 +51,97 @@ export const LaLetterBuilderSendOptions = MiddleProgressStep((props) => {
...
+                </p>
+                {session.landlordDetails?.email && (
+                  <p>{`${li18n._(t`We will email your letter to:`)} ${
+                    session.landlordDetails.email
+                  }`}</p>
+                )}
+              </>
+            ) : (
+              <>
+                <p>
```
Comment (samaratrilling) [inline frontend/lib/laletterbuilder/letter-builder/send-options.tsx:108]: same here

File: frontend/lib/laletterbuilder/letter-builder/send-options.tsx [hunk 2]
```diff
@@ -44,39 +51,97 @@ export const LaLetterBuilderSendOptions = MiddleProgressStep((props) => {
...
+      />
     </Page>
   );
 });
+
+export const ConfirmModal: React.FC<{
+  nextStep: string;
+}> = (props) => {
+  const { session } = useContext(AppContext);
+  const userWillMail =
```
Comment (samaratrilling) [inline frontend/lib/laletterbuilder/letter-builder/send-options.tsx:77]: Should just add a note here to make the habitability bit generalizable to other letter types

File: laletterbuilder/letter_sending.py [hunk 3]
```diff
@@ -130,11 +132,10 @@ def send_letter(letter: models.Letter):
     letter_type = letter.get_letter_type()  # TODO: localize this somewhere
     ld = user.landlord_details
 
-    # TODO: fill in user pref
     if ld.email:
```
Comment (samaratrilling) [inline laletterbuilder/letter_sending.py:135]: And here's where we'll have to check not just if we have a LL email, but if the user wants us to send it. (per our convo with Leslie)

**Review by austensen — APPROVED** [review]
This all looks great! I just noticed some missing spaces in modal text, but otherwise looks good to me!

**Comment by sraby** [comment]
This looks great @shakao! I did a code review and also tested things out locally. There were two thnigs I noticed that might be out of scope for this PR, but wanted to flag just in case:

I noticed that on desktop and mobile, the spacing of the buttons on the modal seemed a bit off:

image

I also noticed that after clicking through the modal, I got redirected back to the "My letters" page. If I clicked "send letter myself," the My Letters page would show the habitability letter still "in progress" and if I clicked "mail for me" I'd be prompted to start a new letter.

**Review by sraby — APPROVED** [review]

**Comment by shakao** [comment]
@sraby thanks for the review! yep, i plan to do a separate PR with that's just an overall CSS pass to make sure the send options page + modal both match the mocks and yeah, the confirmation page will also be separate.

**Pull request merged**